test/functionalTest/cases/3122_location_metadata_issues/create_and_get_location_metadata_v1_v2.test

```diff
@@ -0,0 +1,256 @@
+# Copyright 2018 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Create 2 entities, both with an attribute containing a location metadata,
+One by NGSIV1 POST (of marking location), the other one by NGSIV2 POST.
+See that getting entities both in NGSIV1 and NGSIV2, the attribute metadatas are correctly rendered.
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
+# 01. Create E1 with NGSIV2
+# 02. Create E2 with NGSIV1
+# 03. Get E1 and E2 with NGSIV2
+# 04. Get E1 and E2 with NGSIV1
+
+echo "01. Create E1 with NGSIV2"
+echo "========================="
+payload='
+{
+   "id": "E1",
+   "type": "T",
+   "A": {
+      "value": "40, -3",
+      "type": "whatever",
+      "metadata": {
+         "location": {
+            "type": "string",
+            "value": "WGS84"
+         }
+      }
+   }
+}
+'
+orionCurl --url '/v2/entities' --payload "$payload"
+echo
+echo
+
+
+echo "02. Create E2 with NGSIV1"
+echo "========================="
+payload='
+{
+    "attributes": [
+        {
+            "name": "A",
+            "type": "whatever",
+            "value": "39,-2",
+            "metadatas": [
+               {
+                  "name": "location",
+                  "type": "string",
+                  "value": "WGS84"
+               }
+            ]
+        }
+    ]
+}
+'
+orionCurl --url '/v1/contextEntities/type/T/id/E2' --payload "$payload"
+echo
+echo
+
+
+echo "03. Get E1 and E2 with NGSIV2"
+echo "============================="
+orionCurl --url /v2/entities
+echo
+echo
+
+
+echo "04. Get E1 and E2 with NGSIV1"
+echo "============================="
+orionCurl --url /v1/contextEntities
+echo
+echo
+
```

Comment:
I'd suggest to add a step 5 with a geo query covering both (40, -3) and (39,-2), The result, as far as I understand, should be E2 but not E1 as metadata-based location is not supported in NGSIv2 so "location" there is a regular metadata.

Let's add the step and see what's happen :)

Reply:
Fixed in 1fa60ea